### DISCUSSION TITLE
Redirect request from the External Frontend to an Internal Web Page like Grafana Dashboard running on internal IP/Port

### DISCUSSION THREAD

**Discussion by cee-mnitin** [fastapi/fastapi#6250] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# No Code to share
```

### Description

I have a FastAPI server for a React Frontend. I also have Grafana running on the same server for monitoring purposes. This Grafana server is only on the local network and not exposed on the external IP. 

I want to see the Grafana Dashboard in my React App. So when I send a request to an endpoint, for example, /grafana, it should redirect to INTERNAL_IP:PORT of grafana and display that web page in the frontend. 

I cannot use Redirect as that needs the Grafana IP and Port to be accessible directly from the external Frontend. 

Would be grateful for any suggestions.

### Operating System

Linux

### Operating System Details

Ubuntu 20.04

### FastAPI Version

0.65.2

### Python Version

3.8

**Comment by JarroVGIT** [comment, marked as answer]
IT is not possible directly, as you can’t proxy a request to an internal server and serve it as a webpage of your FastAPI. However, if it is static (and there are no websockets or other requests initiated on the grafana webpage), you could fetch its raw html and return that via your API. 

Keep in mind that any css and JavaScript files in the raw html might still refer to localIP:localPort/css/style.css so you would need countermeasures for that. For example, fetch those in different endpoints of your API and replace references in the raw HTML response to reference to those endpoints. 

IT would be ugly but would work.

**Comment by cee-mnitin** [comment]
I guess I will have to expose the port.

Thanks.
